kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorIT.java

```diff
@@ -0,0 +1,240 @@
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   */
+  private void createS3RootClient() {
+    s3RootClient = AmazonS3ClientBuilder.standard()
+      .withCredentials(
+          new AWSStaticCredentialsProvider(
+              new BasicAWSCredentials(
+                System.getenv("ROOT_USER_ACCESS_KEY_ID"),
+                System.getenv("ROOT_USER_SECRET_ACCESS_KEY"))))
+      .withRegion("ap-south-1")
+      .build();
+  }
+
+  @After
+  public void close() {
+    stopConnect();
+  }
+
+  /**
+   * Success scenario : End to end test
+   * @throws Throwable
+   */
+  @Test
+  public void testToAssertConnectorAndDestinationRecords() throws Throwable {
+    String bucketName = "conn-dest-bucket1";
+    // create bucket
+    createS3Bucket(bucketName);
+    // create topics in Kafka
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+    // send records to kafka
+    sendRecordsToKafka();
+
+    Map<String, String> props = getProperties();
+    props.put("s3.bucket.name", bucketName);
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    waitForConnectorToCompleteSendingRecords(totalNoOfRecordsProduced, FLUSH_SIZE, bucketName);
+
+    // assert records
+    assertEquals(getNoOfObjectsInS3(bucketName), totalNoOfRecordsProduced/FLUSH_SIZE);
+
+    // delete the bucket
+    deleteBucket(bucketName);
+  }
+
+  /**
+   * Case in which bucket permissions/policies are changed while uploading records
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   * Blocker : Assert Condition
+   * @throws Exception
+   */
+  @Test
+  public void testIfBucketPermissionIsChangedWhileUploading() throws Exception {
+    String bucketName = "conn-dest-bucket4";
+
+    createS3Bucket(bucketName);
+    addReadWritePolicyToBucket(bucketName);
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+    // send records to kafka
+    sendRecordsToKafka();
+
+    Map<String, String> props = getProperties();
+    props.put("s3.bucket.name", bucketName);
+    props.put("aws.access.key.id", System.getenv("SECONDARY_USER_ACCESS_KEY_ID"));
+    props.put("aws.secret.access.key", System.getenv("SECONDARY_USER_SECRET_ACCESS_KEY"));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    Thread.sleep(5000);
+
+    // revoke read/write permission
+    s3RootClient.deleteBucketPolicy(bucketName);
+
+    // produce more records to kafka
+    sendRecordsToKafka();
+    Thread.sleep(10000);
```

Comment:
replace withwaitForCondition(...).
Assert no new records should be written to the bucket.

Reply:
We should delete the bucket as well.
Or maybe add that as an @after action to all tests.

Reply:
need to discuss

Reply:
updated test case